### PR TITLE
added functionality to prototype the company contacts data to the inv…

### PR DESCRIPTION
**Pull request by yozzo**
…estment page

### CODE REVIEW

File: src/apps/investment-projects/views/_details-form.njk [hunk 1]
```diff
@@ -126,27 +126,42 @@
...
-          value: client_contact
-        }) }}
-      {% endfor %}
+        <p class="c-form-fieldset__add-another">
+          <button class="button button--secondary js-AddItems__add" type="submit" name="add-item" value="client_contacts" data-persist-values="true">Add another contact</button>
+        </p>
+      </div>
+
+    {% else %}
+      <h2>Pisica</h2>
```
Comment (teneightfive) [inline src/apps/investment-projects/views/_details-form.njk]: What's the purpose of these headings?

File: src/apps/investment-projects/views/_details-form.njk [hunk 2]
```diff
@@ -126,27 +126,42 @@
...
+      <h3>{{ form.state.company_contacts.length > 0  }}</h3>
+      <h3>{{ not form.errors.messages.client_contacts  }}</h3>
+      <h3>{{ form.state.company_contacts.length > 0 and not form.errors.messages.client_contacts  }}</h3>
+      <h3>{{ form.hiddenFields.investor_company }}</h3>
+
+      <p>This company has no contacts, please create a business contact</p>
```
Comment (teneightfive) [inline src/apps/investment-projects/views/_details-form.njk]: Maybe use the [Message component](https://datahub-beta2.herokuapp.com/components/messages) for this as it's a pattern we've [elsewhere](https://github.com/uktrade/data-hub-frontend/blob/5d97cd03fafc842492c6adc238da9364b777066d/src/apps/omis/apps/create/views/contact.njk#L7-L9) to display messages like this.

File: src/apps/investment-projects/views/_details-form.njk [hunk 3]
```diff
@@ -126,27 +126,42 @@
...
+      <a class="button button--secondary" href="/contacts/create?company={{ form.hiddenFields.investor_company }}">Add Contact</a>
+      <a class="button button--secondary" href="{{ form.options.returnLink }}">Return Link</a>
```
Comment (teneightfive) [inline src/apps/investment-projects/views/_details-form.njk]: Do we need the return link?

File: src/apps/investment-projects/views/_details-form.njk [hunk 4]
```diff
@@ -126,27 +126,42 @@
...
+    {% endif %}
+
+    {{ form | log | safe }}
```
Comment (teneightfive) [inline src/apps/investment-projects/views/_details-form.njk]: Maybe remove this dump of the form object.

File: assets/javascripts/modules/add-items.js [hunk 5]
```diff
@@ -65,8 +65,11 @@ const AddItems = {
   cacheEls () {
     const lastItem = this.wrapper.querySelector(this.settings.itemSelector)
 
-    this.itemContainer = lastItem.parentNode
-    this.template = lastItem.cloneNode(true)
+    if (lastItem) {
```
Comment (teneightfive) [inline assets/javascripts/modules/add-items.js]: What's the purpose of this `if` block? In theory we should always be rendering at least one item.

And if we don't maybe we need to handle the template not being there in the `render` method too.

File: src/apps/investment-projects/middleware/forms/details.js [hunk 6]
```diff
@@ -46,6 +51,13 @@ async function populateForm (req, res, next) {
     const state = assign({}, {
       client_contacts: [''],
       business_activities: [''],
+      company_contacts: () => {
+        if (equityCompanyData && equityCompanyData.contacts) {
+          return equityCompanyData.contacts
```
Comment (teneightfive) [inline src/apps/investment-projects/middleware/forms/details.js]: Lodash's [flatten](https://lodash.com/docs/4.17.4#flatten) method might allow this to not require the if/else block.

```js
...
company_contacts: flatten([equityCompanyData.contacts]),
...
```

Also is this being used anywhere in the form? Adding it to the `state` object doesn't feel quite right as it's not holding form state.

If it's to provide the logic in the view `equityCompany.contacts` will already be available in the view as it's set on the `res.locals` object in previous middleware.

File: src/apps/investment-projects/views/_details-form.njk (hunk 4 above)
Comment (yozzo) [reply src/apps/investment-projects/views/_details-form.njk]: work in progress, sir. had to pick up something else, this isn't yet ready for reviewing, therefore didn't attach any reviewers to it. but thank you.

Comment (teneightfive) [reply src/apps/investment-projects/views/_details-form.njk]: Ah my bad. Sorry, was just going through all the PRs at the time.

**Pull request closed without merging**